Parse/src/main/java/com/parse/ParseFileController.java

```diff
@@ -56,6 +56,13 @@ public File getCacheFile(ParseFile.State state) {
     return new File(cachePath, state.name());
   }
 
+  /* package for tests */ File getTempFile(ParseFile.State state) {
+    if (state.url() == null) {
+      return null;
+    }
+    return new File(cachePath, ParseDigestUtils.md5(state.url()));
```

Comment:
Here I just follow what we do for iOS, but I think state.url + (.tmp) is fine. I will use this.
Clear all tmp before re-download make sense, will do that.